docs/LoaderInterfaceArchitecture.md

```diff
@@ -0,0 +1,694 @@
+more optimized to the specific device, but the returned function pointers will
+**only** work for the device used when querying them.
+Also, unlike `vkGetInstanceProcAddr`, `vkGetDeviceProcAddr` can only be used on
+Vulkan device functions.
+
+The best solution is to query instance extension functions using
+`vkGetInstanceProcAddr`, and to query device extension functions using
+`vkGetDeviceProcAddr`.
+See
+[Best Application Performance Setup](LoaderApplicationInterface.md#best-application-performance-setup)
+section in the
+[LoaderApplicationInterface.md](LoaderApplicationInterface.md) document for more
+information on this.
+
+##### Device Extensions
+
+As with instance extensions, a device extension is a set of Vulkan device
+functions extending the Vulkan language.
+More information about device extensions can be found later in this document.
+
+
+### Dispatch Tables and Call Chains
+
+Vulkan uses an object model to control the scope of a particular action or
+operation.
+The object to be acted on is always the first parameter of a Vulkan call and is
+a dispatchable object (see Vulkan specification section 2.3 Object Model).
+Under the covers, the dispatchable object handle is a pointer to a structure,
+which in turn, contains a pointer to a dispatch table maintained by the loader.
+This dispatch table contains pointers to the Vulkan functions appropriate to
+that object.
+
+There are two types of dispatch tables the loader maintains:
+ - Instance Dispatch Table
+   - Created in the loader during the call to `vkCreateInstance`
+ - Device Dispatch Table
+   - Created in the loader during the call to `vkCreateDevice`
+
+At that time the application and the system can each specify optional layers to
+be included.
+The loader will initialize the specified layers to create a call chain for each
+Vulkan function and each entry of the dispatch table will point to the first
+element of that chain.
+Thus, the loader builds an instance call chain for each `VkInstance` that is
+created and a device call chain for each `VkDevice` that is created.
+
+When an application calls a Vulkan function, this typically will first hit a
+*trampoline* function in the loader.
+These *trampoline* functions are small, simple functions that jump to the
+appropriate dispatch table entry for the object they are given.
+Additionally, for functions in the instance call chain, the loader has an
+additional function, called a *terminator*, which is called after all enabled
+layers to marshall the appropriate information to all available implementations.
+
+
+#### Instance Call Chain Example
+
+For example, the diagram below represents what happens in the call chain for
+`vkCreateInstance`.
+After initializing the chain, the loader will call into the first layer's
+`vkCreateInstance` which will call the next finally terminating in the loader
+again where this function calls every implementation's `vkCreateInstance` and
+saves the results.
```

Comment:
```suggestion
After initializing the chain, the loader calls into the first layer's
`vkCreateInstance`, which will call the next layer's `vkCreateInstance 
before finally terminating in the loader again where it will call
every implementation's `vkCreateInstance`.
```